ospd/ospd.py

```diff
@@ -777,9 +777,13 @@ def handle_client_stream(self, stream, is_unix=False):
         while True:
             try:
                 if is_unix:
-                    data.append(stream.recv(1024))
+                    buf = stream.recv(1024)
                 else:
-                    data.append(stream.read(1024))
+                    buf = stream.read(1024)
+                if len(buf):
```

Comment:
I would change this into
```suggestion
                if not buf:
                    # remote has closed the connection
                    break

                data.append(buf)
```

Reply:
Indeed, I have updated the PR. Upon closer inspection I have also removed the superfluous conditional following it.